### PR TITLE
PIM-3564: use ChannelInterface

### PR DESCRIPTION
**Pull request by jjanvier**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | N
| CI currently passes? | Y
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | Y
| Fixed tickets        | -
| DB schema updated?   | -
| Migration script?    | -
| Doc PR               |  -

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php [hunk 1]
```diff
@@ -89,10 +89,10 @@ public function generateMissing()
      * Generate missing completenesses for a channel if provided or a product
      * if provided. CAUTION: the product must be already flushed to the DB
      *
-     * @param ProductInterface $product
-     * @param Channel          $channel
+     * @param ProductInterface  $product
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php]: not aligned

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php [hunk 2]
```diff
@@ -313,13 +313,16 @@ protected function getFieldsNames(array $channels, array $familyReqs)
      * Get the name of a normalized data field
      *
      * @param AttributeInterface $attribute
-     * @param Channel            $channel
-     * @param LocaleInterface    $locale
+     * @param ChannelInterface $channel
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php]: same

File: src/Pim/Bundle/CatalogBundle/Factory/AttributeRequirementFactory.php [hunk 3]
```diff
@@ -20,12 +20,12 @@ class AttributeRequirementFactory
      * Create and configure an attribute requirement instance
      *
      * @param AttributeInterface $attribute
-     * @param Channel            $channel
-     * @param boolean            $required
+     * @param ChannelInterface   $channel
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Factory/AttributeRequirementFactory.php]: too many spaeces

File: src/Pim/Bundle/CatalogBundle/Model/ChannelInterface.php [hunk 4]
```diff
@@ -123,4 +123,19 @@ public function getColor();
...
+     */
+    public function getLocaleCodes();
+
+    /**
+     * To string
+     *
+     * @return string
+     */
+    public function __toString();
+
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Model/ChannelInterface.php]: extra blank line

**Pull request merged**